dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -3266,7 +3266,7 @@ bool DynamicDataImpl::move_single_to_complex(const const_single_iterator& it, Dy
 }
 
 bool DynamicDataImpl::move_single_to_complex_i(const const_single_iterator& it,
-                                               DynamicDataImpl* data, const TypeKind treat_as)
+                                               DynamicDataImpl* data, TypeKind treat_as)
```

Comment:
The function definition can use `const` here, just not in the header.  I think there's a note on this in the gudelines, and if not then we can add one.

Reply:
The guidelines have a related guide (https://opendds.readthedocs.io/en/latest-release/internal/dev_guidelines.html#language-usage, point 6th) but it doesn't explicitly say when to use `const`.

Reply:
I updated the guidelines for this.